include/openssl/evp.h

```diff
@@ -691,20 +691,23 @@ OPENSSL_EXPORT int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx);
 // containing the result. It returns one on success or zero on error.
 OPENSSL_EXPORT int EVP_PKEY_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY **out_pkey);
 
-OPENSSL_EXPORT int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+// EVP_PKEY_encapsulate is operation defined for a KEM (Key Encapsulation
+// Mechanism). The function generates a shared secret and a ciphertext which
+// encapsulates the shared secret using the public key given within |ctx|.
 OPENSSL_EXPORT int EVP_PKEY_encapsulate(EVP_PKEY_CTX *ctx,
```

Comment:
Return values?
Where are the "ciphertext" and "shared secret" output?
What is `secret_key_len` exactly?

I think we might as well write it.